projects/ui-components/lf-user-feedback/feedback-image-upload/feedback-image-upload.component.spec.ts

```diff
@@ -0,0 +1,173 @@
+    expect(fileDropZone.length).toBe(1);
+  });
+
+  it('if there is image attached, should show the picked file zone', () => {
+    // Act
+    component.imageUploaded = new File([''], 'test.png');
+    fixture.detectChanges();
+
+    // Assert
+    const fileDropZone = document.getElementsByClassName('picked-file-zone');
+    expect(fileDropZone.length).toBe(1);
+  });
+
+  it('if tryReadAndValidateImageAsync is called with a valid image, should should attach image, and should emit event feedbackImageBase64', fakeAsync(async () => {
+    // Arrange
+    const base64Image =
+      'iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==';
+    // convert base64 to byte array
+    const file = base64ToImage(base64Image);
+    spyOn(component.feedbackImageBase64, 'emit');
+
+    // Act
+    // @ts-ignore
+    const success = await component.tryReadAndValidateImageAsync(file);
+    fixture.detectChanges();
+
+    // Assert
+    expect(success).toBe(true);
+    expect(component.feedbackImageBase64.emit).toHaveBeenCalledOnceWith(base64Image);
+    expect(component.imageUploaded).toBe(file);
+  }));
+
+  it('if tryReadAndValidateImageAsync is called with a file above 2.9MB, should emit warning, and should not attach image', fakeAsync(async () => {
+    // Arrange
+    const fileSize = 3 * Math.pow(1024, 2); // 3 MB
+    const file = createInvalidImageWithSize(fileSize);
+    spyOn(component.imageUploadError, 'emit');
+
+    // Act
+    // @ts-ignore
+    await component.tryReadAndValidateImageAsync(file);
+    fixture.detectChanges();
+
+    // Assert
+    expect(component.imageUploadError.emit).toHaveBeenCalledWith(
+      'Image not attached. The image exceeds the maximum file size of 2.9 MB.'
+    );
+    expect(component.imageUploaded).toBe(undefined);
+  }));
+
+  it('if tryReadAndValidateImageAsync is called with an invalid image, should emit warning, and should not attach image', fakeAsync(async () => {
+    // Arrange
+    const fileSize = Math.pow(1024, 2); // 1 MB
+    const file = createInvalidImageWithSize(fileSize);
+    spyOn(component.imageUploadError, 'emit');
+
+    // Act
+    // @ts-ignore
+    await component.tryReadAndValidateImageAsync(file);
+    fixture.detectChanges();
+
+    // Assert
+    expect(component.imageUploadError.emit).toHaveBeenCalledWith(
+      'Image not attached. The image is corrupted or in an unrecognized format. The accepted formats are: JPEG, PNG, GIF, WebP.'
+    );
+    expect(component.imageUploaded).toBe(undefined);
+  }));
+
+  it('if multiple images are dropped to the file drop zone, should emit warning', () => {
+    const file1 = new File([''], 'dummy1.png');
+    const file2 = new File([''], 'dummy2.png');
+    const dataTransfer = new DataTransfer();
+    dataTransfer.items.add(file1);
+    dataTransfer.items.add(file2);
+    // @ts-ignore
+    spyOn(component, 'tryReadAndValidateImageAsync').and.callThrough();
```

Comment:
According to Doc: "By chaining the spy with and.callThrough, the spy will still track all calls to it but in addition it will delegate to the actual implementation." We don't need the callThrough here.